### PR TITLE
ENH: pyami

### LINKED ISSUE #33: Filter and Monitor Dets
Old python connected to pyami to set the filter string for the l3 trigger.
Old python got back data from pyami.
Old python normalized the data from pyami by the filter det.

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Add `AmiDet` `Device` that uses its `prefix` to do the `pyami` connection.
- l3t configuration
- add pyami to the `pcdsdaq_lib_setup`

## Motivation and Context

- pyami support was requested
- closes #33 

## How Has This Been Tested?

Added tests that pass for full coverage and found a lot of rough edges in real sessions.

## Where Has This Been Documented?

I added documentation to sphinx

### CODE REVIEW

**Comment by ZLLentz** [comment]
rough edges are:
- only gets 3/4 of events (???)
- need to provide `guess_ami_proxy` somewhere from old `blutil.guessAmiProxy`
- I don't actually know how to test the l3t stuff

**Comment by ZLLentz** [comment]
- 3/4 events went away, probably daq config related
- added get_ami_proxy
- tested l3t stuff

TODO:
- fix handling of det filter string vs pyami filter string
- check up on the correct defaults

**Comment by ZLLentz** [comment]
- fixed handler of det filter string

TODO:
- check up on correct defaults
- look into how much of the config can be automatic
- look into how we're going to activate the daq libraries
- add a set_filter on a per-detector basis

**Comment by ZLLentz** [comment]
TODO:
- fix tests that I just broke
- check up on correct defaults

**Comment by ZLLentz** [comment]
TODO:
- Finish test work
- Correct the defaults based on Silke's input
- Add the monitor det functionality (normalize one detector by another)

File: docs/source/ami_basic.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
+Basic Usage
+===========
+The `AmiDet` class can be used to read the mean and rms of an ami variable
```
Comment (teddyrendahl) [inline docs/source/ami_basic.rst]: I think ReSt want two backquotes -> ``name``?

File: docs/source/ami_basic.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    from pcdsdaq.ami import AmiDet
+
+    det = AmiDet('AMI-NAME', name='ami_det')
+
+
+In a Scan
+---------
+These `AmiDet` objects should be added to the ``dets`` list of a ``bluesky``
+scan. If used as configured above, they will collect data at each scan step
+while the daq is running. Below is an example scan, see the `plans_basic` page
```
Comment (teddyrendahl) [inline docs/source/ami_basic.rst]: missing :ref:?

File: docs/source/ami_basic.rst (hunk 1 above)
Comment (ZLLentz) [reply docs/source/ami_basic.rst]: one backquote hits the directive assigned as `default_role` in `conf.py`, which in this case is the `any` directive, so I backquote like this as a shorthand for linking to the API page

File: docs/source/ami_basic.rst (hunk 2 above)
Comment (ZLLentz) [reply docs/source/ami_basic.rst]: see above, default role is `any`

**Pull request merged**